server/api_test.go

```diff
@@ -1 +1,60 @@
 package server
+
+import (
+	"context"
+	"testing"
+
+	"golang.org/x/crypto/bcrypt"
+
+	"github.com/coreos/dex/api"
+	"github.com/coreos/dex/storage/memory"
+)
+
+var s = memory.New()
+var serv = NewAPI(s)
+
+func TestCreatePassword(t *testing.T) {
+	ctx := context.Background()
+	p := api.Password{
+		Email:    "test@example.com",
+		Username: "test",
+		UserId:   "test123",
+	}
+	hash, err := bcrypt.GenerateFromPassword([]byte("test1"), 10)
+	if err != nil {
+		t.Errorf("Unable to generate password hash: %v", err)
+	}
+	p.Hash = hash
+
+	req := api.CreatePasswordReq{
+		Password: &p,
+	}
+
+	if _, err := serv.CreatePassword(ctx, &req); err != nil {
+		t.Errorf("Unable to create password: %v", err)
+	}
+}
+
+func TestUpdatePassword(t *testing.T) {
+	ctx := context.Background()
+	req := api.UpdatePasswordReq{
+		Email:       "test@example.com",
+		NewUsername: "test1",
+	}
+
+	if _, err := serv.UpdatePassword(ctx, &req); err != nil {
+		t.Errorf("Unable to update password: %v", err)
+	}
+}
+
+func TestDeletePassword(t *testing.T) {
```

Comment:
I commented on this above, but two tests shouldn't ever be dependent on one another. If they are, it should probably just be one test :)